RNF34.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __init__(self, sequence):
+        pass
+
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, slc):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        pass
+
+    @abstractmethod
+    def is_valid_alphabet(self):
+        pass
+
+
+class NucleicAcid(BiologicalSequence):
+    def __init__(self, sequence):
+        self.sequence = sequence
+
+    def __len__(self):
+        return len(self.sequence)
+
+    def __getitem__(self, slc):
+        return self.sequence[slc]
+
+    def __str__(self):
+        return str(self.sequence)
+
+    def __repr__(self):
+        return self.sequence
+
+    def is_valid_alphabet(self):
+        alphabet = type(self).ALPHABET
+        if set(self.sequence).issubset(alphabet):
+            return True
+        else:
+            return False
+
+    def complement(self):
+        if type(self) == NucleicAcid:
+            raise NotImplementedError("Cannot complement NucleicAcid instance")
+
+        map_dict = type(self).MAP
+        comp_seq = "".join([map_dict[base] for base in self.sequence])
+
+        return type(self)(comp_seq)
+
+    def gc_content(self):
+        if type(self) == NucleicAcid:
+            raise NotImplementedError("Cannot gc_content NucleicAcid instance")
+        gc_count = sum([1 for base in self.sequence if base in ["C", "G"]])
+        gc_content = (gc_count / len(self)) * 100
+
+        return gc_content
+
+
+class DNASequence(NucleicAcid):
+    ALPHABET = set("ATGC")
+    MAP = {"A": "T", "T": "A", "C": "G", "G": "C"}
+
+    def transcribe(self):
+        transcribed = self.sequence.replace("T", "U")
+
+        return RNASequence(transcribed)
+
+
+class RNASequence(NucleicAcid):
```

Comment:
Классы DNASequence и RNASequence содержат все необходимые методы, определённые по условию задачи. Однако, в них не определена проверка на соответствие последовательности классу. Грубо говоря, конструкция RNASequence('ATGC') отлично работает, но это биологически не корректно. 

Ранее мы определили метод is_valid_alphabet(), который отлично бы использовался в данных целях. Для того, чтобы его реализовать мы можем переопределить init нового класса. Например, я сделала так:
```suggestion
class RNASequence(NucleicAcid):
        def __init__(self, seq):
            super().__init__(seq)
            if not (super().is_valid_alphabet()):
                raise ValueError('The sequence does not correspond to RNA')
```